app/components/search_status_component.rb

```diff
@@ -0,0 +1,47 @@
+class SearchStatusComponent < ApplicationComponent
+  attr_reader :developer
+
+  def initialize(developer)
+    @developer = developer
+  end
+
+  def render?
+    developer.actively_looking? || developer.open? || developer.not_interested?
+  end
+
+  def icon_tag
+    inline_svg_tag "icons/solid/#{icon}.svg", class: icon_classes.join(" ")
+  end
+
+  def text_classes
+    if developer.actively_looking?
+      "text-green-700"
+    elsif developer.open? || developer.not_interested?
+      "text-gray-900"
+    end
+  end
```

Comment:
The below is an alternative that would ensure `text-gray-900` is used [as a default] if any further statuses were added: 

```suggestion
  def text_classes
    return "text-green-700" if developer.actively_looking?

    "text-gray-900" # default colour unless actively looking 
  end

Reply:
Good call again. Having defaults is definitely the way to go.